app/Filament/Pages/Settings/NotificationPage.php

```diff
@@ -214,21 +214,30 @@ public function sendTestTelegramNotification()
     {
         $notificationSettings = new (NotificationSettings::class);
 
-        if (count($notificationSettings->telegram_recipients)) {
-            foreach ($notificationSettings->telegram_recipients as $recipient) {
-                \Illuminate\Support\Facades\Notification::route('telegram_chat_id', $recipient['telegram_chat_id'])
-                ->notify(new TelegramNotification('Test notification channel *telegram*'));
+        $bot = \Config::get('telegram.bot');
+
+        if (! empty($bot)) {
```

Comment:
For this we should use "return early" should there be a missing token and use the [blank](https://laravel.com/docs/9.x/helpers#method-blank) helper. Example below...

```php
if (blank($bot)) {
    Notification::make()
        ->title('First you need to add \'TELEGRAM_BOT_TOKEN\' on your .env file')
        ->warning()
        ->send();
}
```

After that we can then go into the logic for checking for telegram recipients. 
```php
if (! empty($notificationSettings->telegram_recipients)) {
    foreach ($notificationSettings->telegram_recipients as $recipient) {
         \Illuminate\Support\Facades\Notification::route('telegram_chat_id', $recipient['telegram_chat_id'])
                    ->notify(new TelegramNotification('Test notification channel *telegram*'));
         }

         Notification::make()
             ->title('Test Telegram notification sent.')
             ->success()
             ->send();
} else {
    Notification::make()
        ->title('You need to add recipients to receive Telegram notifications.')
        ->warning()
        ->send();
}
```